### PR TITLE
[feat] add variations create

### PR DESCRIPTION
**Pull request by chantalchung**
TODO: add some tests! -- tested manually, will create ticket for tests
- add variations create command
- prompts for variables values (with proper types) if optional variables flag is not passed in
- ~~refactor cleanup command to use updated variable values prompt, which returns the correct types now~~
    - keep the cleanup command variable values prompt to use `variableValueStringPrompt`, as the types are handled in `src/utils/refactor/RefactorEngine.ts`

cli-variations-create

JSON variable: 

cli-variations-create-json

### CODE REVIEW

File: src/commands/variations/create.ts [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        if (!variables) {
+            const variablesForFeature = await fetchVariables(this.authToken, this.projectKey, featureKey)
+            const variablePrompts = []
+            for (const variable of variablesForFeature) {
+                if (variable.type === 'Boolean') {
+                    variablePrompts.push(variableValueBooleanPrompt(variable.key))
+                } else if (variable.type === 'Number') {
+                    variablePrompts.push(variableValueNumberPrompt(variable.key))
+                } else {
+                    variablePrompts.push(variableValueStringPrompt(variable.key))
```
[the remark this answers is not in the document]
Comment (chantalchung) [reply src/commands/variations/create.ts:78]: oh yeah I assumed that they were stringified and sent to the api that way? let me test it out

[the remark this answers is not in the document]
Comment (chantalchung) [reply src/commands/variations/create.ts]: I added a prompt for JSON variables. It looks like inquirer only returns string/numbers and not JSON objects? So it validates the input by doing `JSON.parse`, but the command that calls the prompt has to call `JSON.parse` on the returned string "answer" again.

File: src/api/variations.ts [hunk 2]
```diff
@@ -1,5 +1,19 @@
+import { IsString, IsNotEmpty, IsOptional } from 'class-validator'
 import apiClient from './apiClient'
-import { Variation } from './schemas'
+import { Feature, Variation } from './schemas'
+
+export class CreateVariationParams {
```
Comment (aurorawalker) [inline src/api/variations.ts:5]: I believe this class is no longer required given some work @emmawillis has been doing - zodios is supposed to auto-pickup all these validation decorators. But might need to check with her, or possibly she can update this on her ticket.

File: src/commands/variations/create.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (elliotCamblor) [reply src/commands/variations/create.ts]: I was going to say that we may need to wrap the JSON.parse in a try/catch to handle for that, but if we can be sure that whatever comes back from the prompt is JSON parsable, then this should be fine

**Review by elliotCamblor — APPROVED** [review]

File: src/commands/variations/create.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (chantalchung) [reply src/commands/variations/create.ts]: yup there's a `validate` function that checks if the input from the prompt is parsable: https://github.com/DevCycleHQ/cli/pull/152/files#diff-7d50853832a5eeba42906784297b6eb9dd035d5aa8d3352161add7053b6d58ecR87-R101

**Pull request merged**